Description: gfxpayload=keep seems to cause trouble with KMS framebuffers,
 and in any case assuming that efifb is applicable to non-EFI systems seems
 optimistic so this test is bogus.

diff -Nur -x '*.orig' -x '*~' grub2/util/grub.d/10_linux.in grub2.new/util/grub.d/10_linux.in
--- grub2/util/grub.d/10_linux.in	2010-02-04 18:20:42.000000000 -0800
+++ grub2.new/util/grub.d/10_linux.in	2010-02-04 18:21:00.000000000 -0800
@@ -74,14 +74,6 @@
 EOF
   save_default_entry | sed -e "s/^/\t/"
 
-  # Use ELILO's generic "efifb" when it's known to be available.
-  # FIXME: We need an interface to select vesafb in case efifb can't be used.
-  if grep -qx "CONFIG_FB_EFI=y" /boot/config-${version} 2> /dev/null ; then
-    cat << EOF
-	set gfxpayload=keep
-EOF
-  fi
-
   if [ -z "${prepare_boot_cache}" ]; then
     prepare_boot_cache="$(prepare_grub_to_access_device ${GRUB_DEVICE_BOOT} | sed -e "s/^/\t/")"
   fi
